Mathlib/AlgebraicTopology/ModelCategory/Basic.lean

```diff
@@ -61,6 +66,93 @@ namespace ModelCategory
 attribute [instance] categoryWithFibrations categoryWithCofibrations categoryWithWeakEquivalences
   cm1a cm1b cm2 cm3a cm3b cm3c cm4a cm4b cm5a cm5b
 
+section
+
+variable [ModelCategory C]
+
+instance : MorphismProperty.IsWeakFactorizationSystem (trivialCofibrations C) (fibrations C) :=
+  MorphismProperty.IsWeakFactorizationSystem.mk' _ _ (fun {A B X Y} i p hi hp ↦ by
+    rw [mem_trivialCofibrations_iff] at hi
+    rw [← fibration_iff] at hp
+    have := hi.1
+    have := hi.2
+    infer_instance)
+
+instance : MorphismProperty.IsWeakFactorizationSystem (cofibrations C) (trivialFibrations C) :=
+  MorphismProperty.IsWeakFactorizationSystem.mk' _ _ (fun {A B X Y} i p hi hp ↦ by
+    rw [mem_trivialFibrations_iff] at hp
+    rw [← cofibration_iff] at hi
+    have := hp.1
+    have := hp.2
+    infer_instance)
+
+end
+
+section mk'
+
+open MorphismProperty
+
+variable {C} in
+private lemma mk'.cm3a_aux [CategoryWithFibrations C]  [CategoryWithCofibrations C]
+    [CategoryWithWeakEquivalences C]
+    [(weakEquivalences C).HasTwoOutOfThreeProperty]
+    [IsWeakFactorizationSystem (trivialCofibrations C) (fibrations C)]
+    [IsWeakFactorizationSystem (cofibrations C) (trivialFibrations C)] {A B X Y : C}
+    {f : A ⟶ B} {w : X ⟶ Y} [Fibration f] [WeakEquivalence w]
+    (h : RetractArrow f w) : WeakEquivalence f := by
+  have hw := factorizationData (trivialCofibrations C) (fibrations C) w
+  have : (trivialFibrations C).IsStableUnderRetracts := by
+    rw [← cofibrations_rlp]
+    infer_instance
+  have sq : CommSq h.r.left hw.i f (hw.p ≫ h.r.right) := ⟨by simp⟩
+  have hf : fibrations C f := by rwa [← fibration_iff]
+  have : HasLiftingProperty hw.i f := hasLiftingProperty_of_wfs _ _ hw.hi hf
+  have : WeakEquivalence hw.i := by simpa only [weakEquivalence_iff] using hw.hi.2
```

Comment:
```suggestion
  have : WeakEquivalence hw.i := weakEquivalence_iff _|>.mpr hw.hi.2
```
There are many (terminal) `by simpa only [..._iff] using` in the PR that can be replaced by direct terms like this. This saves a few call to the simplifier, and is shorter in most cases.

Reply:
These are not very costly invocations of tactics. Using a term saves a few characters, but I would not say that it improves readibility. I feel it is more a matter of taste...